### PR TITLE
kie-tools#2753: Parameterize the groupId of `quarkus-maven-plugin` in "Kogito images" packages

### LINKED ISSUE #2753: Parameterize the groupId of `quarkus-maven-plugin` in "Kogito images" packages
Per quarkus maven tooling documentation [1], it is suggested to invoke quarkus-maven plugin like:

```
  <groupId>${quarkus.platform.group-id}</groupId>
  <artifactId>quarkus-maven-plugin</artifactId>
  <version>${quarkus.platform.version}</version>
```

I suggest we parametrize the groupdID of quarkus in the images, where it is currently hardcoded to `io.quarkus.platform` (see [2] as an example). Note that the groupId could be either io.quarkus or io.quarkus.platform

[1] https://quarkus.io/guides/maven-tooling
[2] https://github.com/apache/incubator-kie-tools/blob/main/packages/sonataflow-image-common/resources/modules/sonataflow/common/scripts/added/add-extension.sh#L50

### PR DESCRIPTION
**Pull request by rgdoliveira**
Closes https://github.com/apache/incubator-kie-tools/issues/2753

### CODE REVIEW

File: packages/sonataflow-image-common/env/index.js [hunk 1]
```diff
@@ -17,11 +17,20 @@
  * under the License.
  */
 
-const { varsWithName, composeEnv } = require("@kie-tools-scripts/build-env");
+const { varsWithName, getOrDefault, composeEnv } = require("@kie-tools-scripts/build-env");
 
 module.exports = composeEnv([require("@kie-tools/root-env/env")], {
-  vars: varsWithName({}),
+  vars: varsWithName({
+    QUARKUS_PLATFORM_groupId: {
```
Comment (ricardozanini) [inline packages/sonataflow-image-common/env/index.js]:
```suggestion
    SONATAFLOW_IMAGE__quarkusGroupId: {
```

File: packages/sonataflow-image-common/env/index.js [hunk 2]
```diff
@@ -17,11 +17,20 @@
...
+      default: "io.quarkus.platform",
+      description: "Quarkus platform group id.",
+    },
+  }),
   get env() {
-    return {};
+    return {
+      quarkus: {
+        groupId: getOrDefault(this.vars.QUARKUS_PLATFORM_groupId),
```
Comment (ricardozanini) [inline packages/sonataflow-image-common/env/index.js]:
```suggestion
      sonataFlowImage: {
        quarkusGroupId: getOrDefault(this.vars.SONATAFLOW_IMAGE__quarkusGroupId),
```

File: packages/sonataflow-management-console-image-env/env/index.js [hunk 3]
```diff
@@ -20,6 +20,7 @@
 const { varsWithName, getOrDefault, composeEnv } = require("@kie-tools-scripts/build-env");
 
 const rootEnv = require("@kie-tools/root-env/env");
+const sonataflowImageCommonEnv = require("@kie-tools/sonataflow-image-common/env");
 
 module.exports = composeEnv([rootEnv], {
```
Comment (ricardozanini) [inline packages/sonataflow-management-console-image-env/env/index.js]:
```suggestion
module.exports = composeEnv([rootEnv, sonataflowImageCommonEnv], {
```

File: packages/sonataflow-management-console-image-env/env/index.js [hunk 4]
```diff
@@ -43,6 +44,10 @@ module.exports = composeEnv([rootEnv], {
       default: 8080,
       description: "The internal container port.",
     },
+    SONATAFLOW_MANAGEMENT_CONSOLE__quarkusPlatformGroupId: {
+      default: sonataflowImageCommonEnv.env.quarkus.groupId,
+      description: "Quarkus platform group id.",
+    },
```
Comment (ricardozanini) [inline packages/sonataflow-management-console-image-env/env/index.js]:
```suggestion

```

File: packages/sonataflow-management-console-image-env/env/index.js [hunk 5]
```diff
@@ -52,6 +57,7 @@ module.exports = composeEnv([rootEnv], {
         name: getOrDefault(this.vars.SONATAFLOW_MANAGEMENT_CONSOLE__name),
         buildTag: getOrDefault(this.vars.SONATAFLOW_MANAGEMENT_CONSOLE__buildTag),
         port: getOrDefault(this.vars.SONATAFLOW_MANAGEMENT_CONSOLE__port),
+        quarkusGroupId: getOrDefault(this.vars.SONATAFLOW_MANAGEMENT_CONSOLE__quarkusPlatformGroupId),
```
Comment (ricardozanini) [inline packages/sonataflow-management-console-image-env/env/index.js]:
```suggestion
```

File: packages/sonataflow-management-console-image/package.json [hunk 6]
```diff
@@ -28,7 +28,7 @@
...
+    "image:cekit:setup:env": ". ./node_modules/@kie-tools/python-venv/venv/bin/activate && cross-env KOGITO_IMAGE_REGISTRY=$(build-env sonataflowManagementConsoleImageEnv.registry) KOGITO_IMAGE_REGISTRY_ACCOUNT=$(build-env sonataflowManagementConsoleImageEnv.account) KOGITO_IMAGE_NAME=$(build-env sonataflowManagementConsoleImageEnv.name) KOGITO_IMAGE_TAG=$(build-env sonataflowManagementConsoleImageEnv.buildTag) QUARKUS_PLATFORM_GROUPID=$(build-env sonataflowManagementConsoleImageEnv.quarkusGroupId) QUARKUS_PLATFORM_VERSION=$(build-env versions.quarkus) KOGITO_VERSION=$(build-env versions.kogito) SONATAFLOW_MANAGEMENT_CONSOLE_PORT=$(build-env sonataflowManagementConsoleImageEnv.port)"
```
Comment (ricardozanini) [inline packages/sonataflow-management-console-image/package.json]:
```suggestion
    "image:cekit:setup:env": ". ./node_modules/@kie-tools/python-venv/venv/bin/activate && cross-env KOGITO_IMAGE_REGISTRY=$(build-env sonataflowManagementConsoleImageEnv.registry) KOGITO_IMAGE_REGISTRY_ACCOUNT=$(build-env sonataflowManagementConsoleImageEnv.account) KOGITO_IMAGE_NAME=$(build-env sonataflowManagementConsoleImageEnv.name) KOGITO_IMAGE_TAG=$(build-env sonataflowManagementConsoleImageEnv.buildTag) QUARKUS_PLATFORM_GROUPID=$(build-env sonataFlowImage.quarkusGroupId) QUARKUS_PLATFORM_VERSION=$(build-env versions.quarkus) KOGITO_VERSION=$(build-env versions.kogito) SONATAFLOW_MANAGEMENT_CONSOLE_PORT=$(build-env sonataflowManagementConsoleImageEnv.port)"
```

**Comment by rgdoliveira** [comment]
I'm going to apply the same changes in the suggestions for the other images.

File: packages/sonataflow-image-common/env/index.js [hunk 7]
```diff
@@ -17,11 +17,20 @@
  * under the License.
  */
 
-const { varsWithName, composeEnv } = require("@kie-tools-scripts/build-env");
+const { varsWithName, getOrDefault, composeEnv } = require("@kie-tools-scripts/build-env");
 
 module.exports = composeEnv([require("@kie-tools/root-env/env")], {
-  vars: varsWithName({}),
+  vars: varsWithName({
+    SONATAFLOW_IMAGE__quarkusGroupId: {
```
Comment (ricardozanini) [inline packages/sonataflow-image-common/env/index.js]:
```suggestion
    KOGITO_IMAGES_CEKIT_MODULES__quarkusGroupId: {
```

File: packages/sonataflow-image-common/env/index.js [hunk 8]
```diff
@@ -17,11 +17,20 @@
...
+      default: "io.quarkus.platform",
+      description: "Quarkus platform group id.",
+    },
+  }),
   get env() {
-    return {};
+    return {
+      sonataFlowImage: {
+        quarkusGroupId: getOrDefault(this.vars.SONATAFLOW_IMAGE__quarkusGroupId),
```
Comment (ricardozanini) [inline packages/sonataflow-image-common/env/index.js]:
```suggestion
        quarkusGroupId: getOrDefault(this.vars.KOGITO_IMAGES_CEKIT_MODULES__quarkusGroupId),
```

File: packages/sonataflow-image-common/env/index.js [hunk 9]
```diff
@@ -17,11 +17,20 @@
...
+  vars: varsWithName({
+    SONATAFLOW_IMAGE__quarkusGroupId: {
+      default: "io.quarkus.platform",
+      description: "Quarkus platform group id.",
+    },
+  }),
   get env() {
-    return {};
+    return {
+      sonataFlowImage: {
```
Comment (ricardozanini) [inline packages/sonataflow-image-common/env/index.js]:
```suggestion
      kogitoImagesCekitModules: {
```

**Review by ricardozanini — COMMENTED** [review]
Sorry last minute name changing.

File: packages/sonataflow-image-common/env/index.js (hunk 9 above)
Comment (rgdoliveira) [reply packages/sonataflow-image-common/env/index.js]: ok, I will adapt per your suggestions

**Comment by rgdoliveira** [comment]
`CI :: Build / run (ubuntu-latest, 1)` failing complaining that it did not find the `kogitoImagesCekitModules.quarkusGroupId` for `sonataflow-management-console-image `. I will take a look

**Comment by ricardozanini** [comment]
@tiagobento do you know why all these changes in the pnpm-lock file?

**Comment by tiagobento** [comment]
@rgdoliveira Honestly, they shouldn't be there I think. I mean, it looks like pnpm rearranged somethings based on previous changes and it didn't revert it when those changes were undone.

My suggestion is that you revert the changes here and bootstrap again, if I'm not mistaken you won't see any changes in it anymore.

**Comment by rgdoliveira** [comment]
@tiagobento ok, I will remove changes from  pnpm-lock file and bootstrap again

**Comment by rgdoliveira** [comment]
@tiagobento you are right, after bootstraping again there were no changes in pnpm-lock file. I've dropped it from the PR.

**Comment by rgdoliveira** [comment]
CI is finally green :)

**Review by ricardozanini — APPROVED** [review]

**Comment by tiagobento** [comment]
Thank you @rgdoliveira. Looks great now! Merging.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**